src/benchmark_runner.cc

```diff
@@ -142,8 +142,9 @@ void RunInThread(const BenchmarkInstance* b, IterationCount iters,
 
   State st = b->Run(iters, thread_id, &timer, manager,
                     perf_counters_measurement, profiler_manager_);
-  BM_CHECK(st.skipped() || st.iterations() >= st.max_iterations)
-      << "Benchmark returned before State::KeepRunning() returned false!";
+  if (!(st.skipped() || st.iterations() >= st.max_iterations)) {
+    st.SkipWithError("FOUND BUG IN YOUR BENCHMARK");
```

Comment:
it might not be a bug... i mean, i think the user should call "SkipWithMessage" which is a bug if they're not, but maybe that's the message we need to put here.

Reply:
Yes, the message is very much a placeholder, please feel free to suggest better verbiage.

Reply:
"The benchmark didn't run, nor was it explicitly skipped.  Please call 'SkipWithXXX` in your benchmark as appropriate."

or we could just output "WARNING: Benchmark did not run and it's unclear if this is intentional."